Code/Framework/AzToolsFramework/Tests/Input/QtEventToAzInputMapperTests.cpp

```diff
@@ -0,0 +1,535 @@
+        QString keyText = keySequence.toString();
+
+        // QKeySequence seems to uppercase alpha keys regardless of shift-modifier
+        if (modifiers == Qt::NoModifier && keyText.isUpper())
+        {
+            keyText = keyText.toLower();
+        }
+        else if (modifiers != Qt::ShiftModifier)
+        {
+            keyText = QString();
+        }
+
+        return AZStd::string(keyText.toUtf8().data());
+    }
+
+    class QtEventToAzInputMapperFixture
+        : public AllocatorsTestFixture
+        , public AzFramework::InputChannelNotificationBus::Handler
+        , public AzFramework::InputTextNotificationBus::Handler
+    {
+    public:
+        static inline constexpr int TestDeviceIdSeed = 4321;
+
+        void SetUp() override
+        {
+            AllocatorsTestFixture::SetUp();
+
+            m_rootWidget = AZStd::make_unique<QWidget>();
+            m_rootWidget->setFixedSize(WidgetSize);
+            m_rootWidget->move(0, 0);
+
+            m_inputChannelMapper = AZStd::make_unique<AzToolsFramework::QtEventToAzInputMapper>(m_rootWidget.get(), TestDeviceIdSeed);
+
+            // listen for events signaled from QtEventToAzInputMapper and forward to the controller list
+            QObject::connect(m_inputChannelMapper.get(), &AzToolsFramework::QtEventToAzInputMapper::InputChannelUpdated, m_rootWidget.get(),
+                [this]([[maybe_unused]] const AzFramework::InputChannel* inputChannel, QEvent* event)
+                {
+                    const QEvent::Type eventType = event->type();
+
+                    if (eventType == QEvent::Type::MouseButtonPress ||
+                        eventType == QEvent::Type::MouseButtonRelease ||
+                        eventType == QEvent::Type::MouseButtonDblClick)
+                    {
+                        m_signalEvents.push_back(QtEventInfo(static_cast<QMouseEvent*>(event)));
+                        event->accept();
+                    }
+                    else if (eventType == QEvent::Type::Wheel)
+                    {
+                        m_signalEvents.push_back(QtEventInfo(static_cast<QWheelEvent*>(event)));
+                        event->accept();
+                    }
+                    else if (eventType == QEvent::Type::KeyPress ||
+                            eventType == QEvent::Type::KeyRelease ||
+                            eventType == QEvent::Type::ShortcutOverride)
+                    {
+                        m_signalEvents.push_back(QtEventInfo(static_cast<QKeyEvent*>(event)));
+                        event->accept();
+                    }
+                });
+        }
+
+        void TearDown() override
+        {
+            m_inputChannelMapper.reset();
+
+            m_rootWidget.reset();
+
+            AllocatorsTestFixture::TearDown();
+        }
+
+        void OnInputChannelEvent(const AzFramework::InputChannel& inputChannel, bool& hasBeenConsumed) override
+        {
+            ASSERT_FALSE(hasBeenConsumed);
```

Comment:
comment: I've never a fan of having test assertions in other functions, it's usually better to have the assert in a test body so if it fails it's clear the reason why. Not a huge deal and you can leave this as-is but something to keep in mind. Another option is just to make this an `AZ_Assert` if it really should never happen and it's not a test assertion/expectation.